### PR TITLE
Add MLX-LM Text Completion Backend

### PR DESCRIPTION
**Pull request by realyxl**
## Checklist:

- [X] I have read the [Contribution guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

## Background

[MLX](https://opensource.apple.com/projects/mlx/) is Apple's official machine learning framework, purpose-built for Apple Silicon's unified memory architecture. [mlx-lm](https://github.com/ml-explore/mlx-lm) is the companion Python package providing LLM inference, quantization, fine-tuning, and an OpenAI-compatible HTTP server.

Today, Mac users running `mlx_lm.server` connect through **Generic (OpenAI-compatible)**, whose whitelist doesn't cover most of mlx-lm's protocol fields. This PR adds mlx-lm as a first-class text completion source, matching the pattern of other local backends, and supports all mlx-lm protocol fields.

## Implementation

All mlx-lm protocol fields are supported: `model`, `max_completion_tokens`, `max_tokens`, `temperature`, `top_p`, `top_k`, `min_p`, `repetition_penalty`, `presence_penalty`, `frequency_penalty`, `repetition_context_size`, `presence_context_size`, `frequency_context_size`, `num_draft_tokens`, `adapters`, `draft_model`, `seed`, `logit_bias`, `stop`, `stream`, `stream_options`, `logprobs`, `top_logprobs`, `xtc_probability`, `xtc_threshold`.

The integration reuses existing ST controls wherever possible, adds new UI only for mlx-specific fields, and follows the same patterns as other text-completion backends in ST.

A few upstream mlx-lm limitations are gated in code with inline comments and will resolve once the upstream fixes land. Everything else currently works as expected.

### UI screenshots

**API Connections**

Screenshot_apiconnections

**AI Response Configuration**

Screenshot_airesponseconfiguration

## Testing

ESLint reports no new errors. End-to-end testing on M3 Pro covered the full integration surface.

To reproduce locally:

```bash
uv pip install mlx-lm
mlx_lm.server --model <model-id>
```

Then in ST: **Text Completion → MLX-LM**, set API URL to `http://127.0.0.1:8080`, click **Connect**, and send a message.

### CODE REVIEW

**Comment by realyxl** [comment]
Since this PR was opened, both SillyTavern staging and mlx-lm upstream have moved on, so parts of it are out of date: some of the upstream limitations it works around have since been fixed on mlx-lm main, and a few of its own defaults need adjusting.

If there's interest in an MLX-LM backend, I'm happy to rework the PR against the current state of both projects and trim it down. If not, feel free to close it.
